src/components/cylc/Task.vue

```diff
@@ -0,0 +1,176 @@
+      ></circle>
+      <!-- circle in the middle
+             * position in the middle
+             * radius can be changed independently
+      -->
+      <circle
+        id="hub"
+        cx="50" cy="50"
+        r="12">
+      </circle>
+      <!-- outer circle
+             * position in the middle
+             * let radius = 50% - (2*stroke-width)
+      -->
+      <circle
+        id="outline"
+        cx="50" cy="50"
+        r="46"
+        stroke-width="8">
+      </circle>
+      <!-- cross (failure)
+             * create a plus and rotate it 45 deg around the origin
+             * let rx ~= max(width, height) * 0.1
+      -->
+      <g
+        id="cross"
+        transform="rotate(45, 50, 50)"
+      >
+        <rect
+          x=15 y=45
+          width="70" height="10"
+          rx="7" ry="7"
+        ></rect>
+        <rect
+          x=45 y=15
+          width="10" height="70"
+          rx="7" ry="7"
+        ></rect>
+      </g>
+    </svg>
+  </span>
+</template>
+
+
+<style lang="scss">
+    $colour: rgb(90,90,90);
+
+    @mixin disk() {
+        #outline {
+            stroke: $colour;
+            fill: $colour;
+        }
+    }
+
+    @mixin outline() {
+        #outline {
+            stroke: $colour;
+        }
+    }
+
+    @mixin hub() {
+        #hub {
+            fill: $colour;
+        }
+    }
+
+    @mixin progress() {
+        #progress {
+            fill: transparent;
+            stroke: $colour;
+            transform-origin: 50% 50%;
+            transform: rotate(-90deg);
+            opacity: 0.4;
+        }
+    }
+
+    @mixin cross() {
+        #cross rect {
+            fill: $colour;
+        }
+    }
+
+    svg.task {
+        /* scale the icon to the font-size */
+        width: 1em;
+        height: 1em;
+
+        circle, rect {
+            /* if no task status display nothing */
+            fill: transparent;
+            stroke: transparent;
+        }
+
+        &.waiting {
+            @include outline();
+        }
+
+        &.submitted {
+            @include outline();
+            @include hub();
+        }
+
+        &.running {
+            @include outline();
+            @include hub();
+            @include progress();
+        }
+
+        &.succeeded {
+            @include disk();
+        }
+
+        &.failed {
+            @include outline();
+            @include cross();
+        }
+
+        &.submit-failed {
+            @include outline();
+            @include hub();
+            @include cross();
+        }
+    }
+</style>
+
+
+<script>
+    export default {
+        name: 'Task',
+        data: function () {
+            return {
+                'status': 'running',
+                'progress': 25
+            }
+        },
+        computed: {
+            progressStyle: function () {
+                return {
+                    'stroke-dashoffset':
+                        'calc(157 - (157 * ' + this.progress/100 + '))'
```

Comment:
I haven't done JS for a long time, but is it OK to concatenate strings and numbers like this?

Reply:
It's OK and quite common, results are predictable:

```js
'a' + 1
"a1"
'a' + 0.1
"a0.1"
'a' + 0.10
"a0.1"
'a' + null
"anull"
```

However, it is nicer to use the new format strings which one can apparently embed statements in (because JS and Python are getting kinda similar nowerdays).

Reply:
OK. So JS is closer to Perl than Python. 😉